src/devices/ni/KompleteKontrol.cpp

```diff
@@ -243,9 +260,13 @@ enum class KompleteKontrolBase::Button : uint8_t
 
 //--------------------------------------------------------------------------------------------------
 
-KompleteKontrolBase::KompleteKontrolBase()
-  : m_isDirtyLeds(true)
+KompleteKontrolBase::KompleteKontrolBase(const NUM_KEYS numKeys)
+  : m_numKeys(static_cast<unsigned>(numKeys))
+  , m_ledsKeysSize(m_numKeys * 3U)
+  , m_ledsKeys(new uint8_t[m_ledsKeysSize])

```

Comment:
c style array creation, since you're using pointers for data writing I didn't change this to a vector

Reply:
maybe a shared_ptr with the array is an option here